### PR TITLE
[MBL-1210] Post campaign checkout

### PR DESCRIPTION
**Pull request by ifosli**
# 📲 What

Add a mostly completed post campaign checkout page. This page is presented after the confirm details page. See `TODO` section for remaining work.

Note: This (including the TODOs) only update the UI. The payment sheet is not wired up yet at all.

# 👀 See

[Jira](https://kickstarter.atlassian.net/browse/MBL-1210)

image

# ♿️ Accessibility 

- [x] Tap targets use minimum of 44x44 pts dimensions
- [x] Dynamic Type should be fine since this isn't adding any new components 

Note: VoiceOver will be tested later to unblock work.

# ✅ Acceptance criteria

- [x] Checkout page shows up (for rewards that include shipping) and matches the mocks with the exceptions listed below

# ⏰ TODO

TODO in follow-up prs to unblock work
- Update rewards summary table to fully support the checkout page. This requires supporting nil shipping and figuring out why bonus support hides on this page
- Update CTA to a) show "charged immediately" warning, probably by creating a `postCampaignPledge` context and b) properly set fields & respond to login flow
- Add snapshot tests once the above UI issues are fixed

### CODE REVIEW

File: Kickstarter-iOS/Features/PledgePaymentMethods/Controller/PostCampaignPledgeRewardsSummaryViewController.swift [hunk 1]
```diff
@@ -44,6 +44,8 @@ final class PostCampaignPledgeRewardsSummaryViewController: UIViewController {
   override func viewDidLoad() {
     super.viewDidLoad()
 
+    self.view.isHidden = true
+    self.pledgeTotalViewController.view.isHidden = true
```
Comment (ifosli) [inline Kickstarter-iOS/Features/PledgePaymentMethods/Controller/PostCampaignPledgeRewardsSummaryViewController.swift:48]: Note: It's important to never put anything in `bindStyles` that can be overwritten later. Anything to do with initial state needs to go in `viewDidLoad` instead. I spent hours trying to debug why the table wasn't showing up on the checkout screen 😭

File: Kickstarter-iOS/Features/PledgeView/Controllers/PledgeViewControllerHelpers.swift [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import KsApi
+import Library
+import UIKit
+
+struct PledgeViewControllerHelpers {
```
Comment (ifosli) [inline Kickstarter-iOS/Features/PledgeView/Controllers/PledgeViewControllerHelpers.swift:5]: This function is literally copy-pasted from the `pledgeViewController` with no changes; no need to review carefully.

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| Library/ViewModels/ConfirmDetailsViewModel.swift:361:1 | warning: (consecutiveSpaces) Replace consecutive spaces with a single space. |
| Library/ViewModels/ConfirmDetailsViewModelTests.swift:845:1 | warning: (indent) Indent code in accordance with the scope level. |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Kickstarter-iOS/Features/PledgeView/Controllers/PostCampaignCheckoutViewController.swift [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+  private lazy var paymentMethodsViewController = {
+    PledgePaymentMethodsViewController.instantiate()
+    // TODO: Add self as delegate and add support for delegate methods.
+  }()
+
+  private lazy var pledgeCTAContainerView: PledgeViewCTAContainerView = {
+    PledgeViewCTAContainerView(frame: .zero)
+      |> \.translatesAutoresizingMaskIntoConstraints .~ false
+    // TODO: Add self as delegate and add support for delegate methods.
```
Comment (ifosli) [inline Kickstarter-iOS/Features/PledgeView/Controllers/PostCampaignCheckoutViewController.swift:25]: I wasn't sure if we have a ticket for wiring up the payment flow here; let me know if we do and I'll link it

**Review by ifosli — COMMENTED** [review]
Note: The initial prs are neatly organized if you want to review them in order. However, that merge conflict was evil, so I gave up on organized commits after that.

**Review by scottkicks — APPROVED** [review]
GTG one the circleci issues are resolved

File: Kickstarter-iOS/Features/PledgePaymentMethods/Controller/PostCampaignPledgeRewardsSummaryViewController.swift (hunk 1 above)
Comment (ifosli) [reply Kickstarter-iOS/Features/PledgePaymentMethods/Controller/PostCampaignPledgeRewardsSummaryViewController.swift]: Thanks for calling that out! For some reason, the separator view was still showing... I just hid that as well and the table is back to being fully hidden before data loads!

**Pull request merged**